### PR TITLE
Switch to vite for faster builds

### PR DESCRIPTION
**Pull request by sirpy**
# Description

### CODE REVIEW

**Review by johnsmith-gooddollar — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<!DOCTYPE html>
```
Comment (johnsmith-gooddollar) [inline index.html:1]: why this file is inside root ? not in public/ as should be

File: inherits.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = function inherits(ctor, superCtor) {
```
Comment (johnsmith-gooddollar) [inline inherits.js:1]: does it used anywhere now ? in first commits it was referenced from vite config, now I can't find any mentions

File: lingui.config.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// import { formatter } from '@lingui/format-json'
```
Comment (johnsmith-gooddollar) [inline lingui.config.ts:1]: I'd like to move i18n to the separate PR as this isn't related to the main scope

File: lint-staged.config.js [hunk 4]
```diff
@@ -9,23 +9,18 @@ const tsConfigPath = './tsconfig.lint-staged.json'
 const tsConfig = (files) => `
 {
   "extends": "./tsconfig.json",
-  "include": [
```
Comment (johnsmith-gooddollar) [inline lint-staged.config.js:12]: why do you removed this ? it includes all our custom typedefs, without them tsc may fail checking syntax on commit hook

File: lint-staged.config.js [hunk 5]
```diff
@@ -9,23 +9,18 @@ const tsConfigPath = './tsconfig.lint-staged.json'
...
 
 module.exports = {
-  '**/*.{js,jsx,ts,tsx}': async (files) => {
-    const ignored = await Promise.all(files.map(async (file) => eslint.isPathIgnored(file)))
-    const filtered = files.filter((_, index) => !ignored[index]).map((path) => `"${path}"`)
-    
-    await writeFileAsync(tsConfigPath, tsConfig(filtered))
-    return [`eslint --no-ignore --max-warnings=0 ${filtered.join(' ')} --fix`, `tsc --p ${tsConfigPath}`]
-  },
+    'src/**/*.{js,jsx,ts,tsx}': async (files) => {
```
Comment (johnsmith-gooddollar) [inline lint-staged.config.js:19]: have you tested does lint staged works now  ? especially when you commit file referencing to some .d.ts from source or to some file from /types in imports

File: src/index.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import './styles/index.css'
 import '@fontsource/dm-sans/index.css'
 import 'react-tabs/style/react-tabs.css'
-import './bootstrap'
+import './bootstrap.ts'
```
Comment (johnsmith-gooddollar) [inline src/index.tsx:4]: why does it needed ? doesn't builder runs through all possible variants if module extension omitted (e.g. js -> jsx -> ts -> tsx -> mjs etc)

File: vite.config.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        //     template: 'treemap', // or sunburst
+        //     open: true,
+        //     gzipSize: true,
+        //     brotliSize: true,
+        //     filename: 'analice.html',
+        // }) as PluginOption,
+        dynamicImports(), //for lingui dynamic import lang files
+        // checker({
+        //     // e.g. use TypeScript check
+        //     typescript: true,
```
Comment (johnsmith-gooddollar) [inline vite.config.ts:43]: why ts check is disabled ?

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (sirpy) [reply index.html:1]: that's vite requirements

File: inherits.js (hunk 2 above)
Comment (sirpy) [reply inherits.js:1]: yeah can be removed

File: lingui.config.ts (hunk 3 above)
Comment (sirpy) [reply lingui.config.ts:1]: we had to upgrade to v4 for vite. so it requires the fixes here.
i'll see if it is possible to still keep lingui v3

File: lint-staged.config.js (hunk 4 above)
Comment (sirpy) [reply lint-staged.config.js:12]: it wasnt working. not sure why we need to check the typedefs

File: src/index.tsx (hunk 6 above)
Comment (sirpy) [reply src/index.tsx:4]: there was a name conflict. i'll retest if still required

File: vite.config.ts (hunk 7 above)
Comment (sirpy) [reply vite.config.ts:43]: it takes long time. i prefer typechecking is part of commit lint-stage and in code editor

**Comment by johnsmith-gooddollar** [comment]
@sirpy @L03TJ3 

1. need to resolve conflicts
2. when it could be merged ?

File: linguifix.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+// console.log({ catalog, output })
+
+// async main() {
+//   const catalog = jsonFormatter.parse(await fs.readFile('myfile.json', "utf8"));
+//   await fs.writeFile('myfile.po', poFormatter.serialize(newFileConent))
+//   console.log('Done!')
+// }
+
+// main();
```
Comment (johnsmith-gooddollar) [inline linguifix.js:38]:
```suggestion
const { readdir, readFile, writeFile } = require('fs')
const { promisify } = require('util')
const { mapValues } = require('lodash')
const { formatter: createPoFormatter } = require('@lingui/format-po')
const { formatter: createJsonFormatter } = require('@lingui/format-json')

const poFormatter = createPoFormatter()
const jsonFormatter = createJsonFormatter({ style: 'lingui' })
const lFormatter = createJsonFormatter({ style: 'minimal' })

const [readdirAsync, readFileAsync, writeFileAsync] = [readdir, readFile, writeFile].map(promisify)
const [parseJSON, parsePO] = [jsonFormatter, poFormatter].map(formatter => buffer => formatter.parse(buffer))

readdirAsync('src/language/locales')
  .then(async dir => Promise.all(dir.map(async item => {
    const path = 'src/language/locales/item' + item
    const catalog = await readFileAsync(path + '/catalog.json.bak', 'utf8').then(parseJSON)
    const compiledCatalog = await readFileAsync(path + '/catalog.po', 'utf8').then(parsePO)

    const newCatalog = mapValues(compiledCatalog, item => {
      const { message } = item
      const translation = catalog[message]

      return !translation || translation === message ? item : { ...item, translation }
    })
    
    const output = poFormatter.serialize(newCatalog, {})

    await writeFileAsync(path + '/catalog.po', output)
  })))
  .then(() => console.log('done'))

```

**Review by johnsmith-gooddollar — APPROVED** [review]

File: src/language/index.tsx [hunk 9]
```diff
@@ -8,9 +8,6 @@ import { AsyncStorage } from '@gooddollar/web3sdk-v2'
 export const locales = ['de', 'en', 'es-AR', 'es', 'it', 'he', 'ro', 'ru', 'vi', 'zh-CN', 'zh-TW', 'ko', 'ja']
```
Comment (johnsmith-gooddollar) [inline src/language/index.tsx:8]: why do we have russian locale ?

Comment (johnsmith-gooddollar) [reply src/language/index.tsx:8]: I remember that GoodDollar supports people not governments but it's not the case
if you would look any interview with the russian citizens - they totally supports shelling of the cities and civil infrastructure, they wants more shelling, they wants all cities to be totally destroyed.

just go to YouTube and search a bit. if you didn't knew you'll be shocked. that's a totally mind-washed nation now, unfortunately.

**Comment by L03TJ3** [comment]
@sirpy @johnsmith-gooddollar 
Please re-review, latest 2 commits come with the following changes:
1. upgraded tailwind/postcss since it was still relying on a specific CRA compatibility package for postcss (seems fine, and verified working on vercel deployment + local

2. I upgraded typescript to 5>. This was done while I was figuring out a tsc compile type error (preventing commit). In the end, the upgrade would not be necessary since the actual fix was just a more declarative type definition needed in the useActiveOnboard hook. 
But I also have not encountered any new issues besides a few extra type definitions around the tailwind theme object.

I don't see any issues in keeping the upgrade, but maybe one of you has an objection as to why we would not upgrade now
If that's the case, ill revert typescript. No objection? then it could be merged after a review

File: src/language/index.tsx (hunk 9 above)
Comment (L03TJ3) [reply src/language/index.tsx]: done

File: lint-staged.config.js (hunk 5 above)
Comment (L03TJ3) [reply lint-staged.config.js:19]: I guess it works.
last 2 commits were around a file using package @gooddollar/goodprotocol where we have additional typings definined in <root>/types

Or should it be tested in a different way?

**Review by L03TJ3 — APPROVED** [review]

**Comment by sirpy** [comment]
@johnsmith-gooddollar please revert linguifix changes. unless you point to a bug, I only see unrequired refactoring which wasnt first approved by me.

@L03TJ3 @johnsmith-gooddollar please revert the russian removal. G$ is not political and is not taking any sides besides the side of the people, helping all people of the world. Not all russians are brainwashed and not all russian speaking people live in russia or are russians.

**Comment by johnsmith-gooddollar** [comment]
> @L03TJ3 @johnsmith-gooddollar please revert the russian removal. G$ is not political and is not taking any sides besides the side of the people, helping all people of the world. Not all russians are brainwashed and not all russian speaking people live in russia or are russians.

This isn't politics anymore. Do you read news ? They just exploded Kahovka water reservoir. A lot of square kilometres of land, cities and the villages will be flooded. Also, changing of the water level may affect to Zaporizhzhia nuclear plant (a similar way it was at Fukushima). We could receive the second one Tchernobyl !

File: lint-staged.config.js (hunk 5 above)
Comment (johnsmith-gooddollar) [reply lint-staged.config.js:19]: @L03TJ3 yes pls test it explicitly
- change some js,jsx,ts,tsx files from src
- also change some file matching those wildcard '/types/*/*d.ts', '/src/**/*.test.ts*', '/**/*.*js', '*.config.ts'
- do a commit from cmdline
- check which files been added to eslint command

File: package.json [hunk 10]
```diff
@@ -236,8 +235,13 @@
         "react-text-mask": "^5.4.3",
         "rxjs": "^7.5.6",
         "text-mask-addons": "^3.8.0",
+        "typescript": "^5.1.3",
```
Comment (johnsmith-gooddollar) [inline package.json:238]: I'm ok with TS5

File: src/components/SideBar.tsx [hunk 11]
```diff
@@ -55,7 +55,7 @@ export default function SideBar({ mobile, closeSidebar }: { mobile?: boolean; cl
     })
 
     const importToMetamask = async () => {
-        const allTokens = []
+        const allTokens: Array<any> = []
```
Comment (johnsmith-gooddollar) [inline src/components/SideBar.tsx:58]: Array<T> is obsolete notation, we prefer to use T[]

```suggestion
        const allTokens: any[] = []
```

File: src/components/TransactionConfirmationModal/index.tsx [hunk 12]
```diff
@@ -5,7 +5,7 @@ import { Text } from 'rebass'
 import styled, { useTheme } from 'styled-components'
 import Circle from '../../assets/images/blue-loader.svg'
 import { useActiveWeb3React } from '../../hooks/useActiveWeb3React'
-import { ExternalLink } from '../../theme'
+import { ExternalLink, theme as TwTheme } from '../../theme'
```
Comment (johnsmith-gooddollar) [inline src/components/TransactionConfirmationModal/index.tsx:8]: define

```typescript
export type TwTheme =  ReturnType<typeof theme>
```

 at `theme/index.tsx` and use it everywhere

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: src/components/SideBar.tsx (hunk 11 above)
Comment (L03TJ3) [reply src/components/SideBar.tsx]: done

File: src/components/TransactionConfirmationModal/index.tsx (hunk 12 above)
Comment (L03TJ3) [reply src/components/TransactionConfirmationModal/index.tsx]: done

File: lint-staged.config.js (hunk 5 above)
Comment (L03TJ3) [reply lint-staged.config.js:19]: Tested, expected files to lint against

**Review by johnsmith-gooddollar — APPROVED** [review]

**Comment by sirpy** [comment]
> > @johnsmith-gooddollar please revert linguifix changes. unless you point to a bug, I only see unrequired refactoring which wasnt first approved by me.
> 
> You uses a lot of XXXSync fns, I've just changes to async ones, promisified (sync functions should be used in node only if no other way) also performed some cleanup from things not used anymore and some reformatting as it was hard for me to understand the original code

as i've requested countless times, no refactoring unless I aprove it first

**Comment by johnsmith-gooddollar** [comment]
> > > @johnsmith-gooddollar please revert linguifix changes. unless you point to a bug, I only see unrequired refactoring which wasnt first approved by me.
> > 
> > 
> > You uses a lot of XXXSync fns, I've just changes to async ones, promisified (sync functions should be used in node only if no other way) also performed some cleanup from things not used anymore and some reformatting as it was hard for me to understand the original code
> 
> as i've requested countless times, no refactoring unless I aprove it first

it just a small utility
and it was needed at least code cleanup

**Comment by L03TJ3** [comment]
@johnsmith-gooddollar @sirpy
Is the linguifix good now? or should the commit still be reverted
if we can agree on current lingui-fix this pr can be merged
